lib/puppet/provider/graylog_api/graylog_api.rb

```diff
@@ -21,21 +21,38 @@ def self.prefetch(resources)
     password = resources[:api][:password]
     port = resources[:api][:port]
     username = resources[:api][:username]
+    tls = resources[:api][:tls]
+    server = resources[:api][:server]
+    verify_tls = resources[:api][:verify_tls]
+    ssl_ca_file = resources[:api][:ssl_ca_file]
     Puppet::Provider::GraylogAPI.api_password = password
     Puppet::Provider::GraylogAPI.api_port = port
-    Puppet::Provider::GraylogAPI.api_username = username
-    wait_for_api(port)
-    resources[:api].provider = new({password: password, port: port, username: username})
+    Puppet::Provider::GraylogAPI.api_tls = tls
+    Puppet::Provider::GraylogAPI.api_server = server
+    Puppet::Provider::GraylogAPI.verify_tls = verify_tls
+    Puppet::Provider::GraylogAPI.ssl_ca_file = ssl_ca_file
+
+    # Settings needed before connection to the server.  We set them on the global
+    # level, so we only need to define them once here.
+    if tls
+      HTTParty::Basement.default_options.update(verify: verify_tls)
+      # we configure this anyway, even we don't want to verify the certificate
+      HTTParty::Basement.default_options.update(ssl_ca_file: ssl_ca_file)
```

Comment:
I'm a little conflicted on this. On the one hand, it's much more concise and convenient than any alternative I can think of; on the other hand, in the (admittedly unlikely) event that there's another Puppet provider used in the same catalog that also uses HTTParty, won't this interfere with the operation of that provider?

Reply:
Makes it possible to only define these  setting once.  I suspect if we choose not to do it this way, we need to add all the settings to the other providers.

Reply:
All of the other providers in the module inherit from this one, so we should be able to define our options centrally and then take advantage of them throughout the module, without setting process-wide defaults that might interfere with providers in other modules. That said, I'm happy to merge this now and then refactor it myself.